src/main.js

```diff
@@ -103,17 +122,29 @@ createNewBookButton.addEventListener('click', function(event) {
   descriptors.push(userDescriptor2.value)
 
   var userCreatedCover = buildNewCover(userCover.value, userTitle.value, userDescriptor1.value, userDescriptor2.value);
-  savedCovers.unshift(userCreatedCover);
-  console.log(userCreatedCover)
+
+  // savedCovers.unshift(userCreatedCover); //this could cause repeat covers if a user inputs the same cover data into the form twice
 
   updateCover(userCreatedCover);
+  saveVisibleCover();
 
   switchToHomeView();
 
   event.preventDefault();
 
 });
 
-function getRandomIndex(array) {
-  return Math.floor(Math.random() * array.length);
+// **** Save current cover functionality ****
+function saveVisibleCover() {
+  console.log(savedCovers);
+  var isRepeat = false;
+  for (var i = 0; i < savedCovers.length; i++) {
+    if (visibleCover === savedCovers[i]) {
+      isRepeat = true
```

Comment:
Do you want to have some kind of response for the user if it's a duplicate?

Reply:
Oh, good idea. We will include this in iteration 5!!